$MirOS: ports/emulators/qemu/snapshot/patches/patch-hw_nand_c,v 1.1 2008/05/07 15:53:10 tg Exp $
--- hw/nand.c.orig	Wed Sep 23 19:01:08 2009
+++ hw/nand.c	Sat Oct 17 16:35:43 2009
@@ -83,22 +83,29 @@ struct NANDFlashState {
 
 # define PAGE(addr)		((addr) >> ADDR_SHIFT)
 # define PAGE_START(page)	(PAGE(page) * (PAGE_SIZE + OOB_SIZE))
+# undef PAGE_MASK
 # define PAGE_MASK		((1 << ADDR_SHIFT) - 1)
 # define OOB_SHIFT		(PAGE_SHIFT - 5)
 # define OOB_SIZE		(1 << OOB_SHIFT)
 # define SECTOR(addr)		((addr) >> (9 + ADDR_SHIFT - PAGE_SHIFT))
 # define SECTOR_OFFSET(addr)	((addr) & ((511 >> PAGE_SHIFT) << 8))
 
+# undef PAGE_SIZE
+# undef PAGE_SHIFT
 # define PAGE_SIZE		256
 # define PAGE_SHIFT		8
 # define PAGE_SECTORS		1
 # define ADDR_SHIFT		8
 # include "nand.c"
+# undef PAGE_SIZE
+# undef PAGE_SHIFT
 # define PAGE_SIZE		512
 # define PAGE_SHIFT		9
 # define PAGE_SECTORS		1
 # define ADDR_SHIFT		8
 # include "nand.c"
+# undef PAGE_SIZE
+# undef PAGE_SHIFT
 # define PAGE_SIZE		2048
 # define PAGE_SHIFT		11
 # define PAGE_SECTORS		4
